### PR TITLE
DOCS: Add docs about CI benchmarking

### PR DESCRIPTION
**Pull request by MridulS**
This fixes https://github.com/astropy/astropy/issues/16216

Maybe we should also add these docs to https://docs.astropy.org/en/latest/development/workflow/maintainer_workflow.html?

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
Thanks! Where in maintainer workflow docs you have in mind?

File: docs/development/testguide.rst
Comment (pllim) [inline docs/development/testguide.rst:1083]:
```suggestion
action only runs if a maintainer adds the label ``benchmark`` to the pull request.
```

To avoid `docs/development/testguide.rst:1080: WARNING: py:obj reference target not found: benchmark`

File: docs/development/testguide.rst
Comment (mhvk) [inline docs/development/testguide.rst:1084]: "run on virtual machines (and thus shared hardware) but ..."

**Comment by eerovaher** [comment]
The added documentation looks reasonable, but (like @MridulS already suggested) I would also expect this to be documented in https://github.com/astropy/astropy/blob/main/docs/development/workflow/maintainer_workflow.rst

File: docs/development/workflow/maintainer_workflow.rst
Comment (pllim) [inline docs/development/workflow/maintainer_workflow.rst:186]:
```suggestion
performance implications. The benchmark action takes significantly longer than regular CI to
```

File: docs/development/workflow/maintainer_workflow.rst
Comment (pllim) [inline docs/development/workflow/maintainer_workflow.rst:189]: Which one is more correct? 🤷‍♀️ 
```suggestion
It is important to note that the benchmarking on Github Actions will be flaky and
```

File: docs/development/workflow/maintainer_workflow.rst
Comment (pllim) [inline docs/development/workflow/maintainer_workflow.rst:183]:
```suggestion
in the Actions tab. This workflow uses the benchmarks
from `Astropy Benchmarks <https://github.com/astropy/astropy-benchmarks/>`_.
```

File: docs/development/workflow/maintainer_workflow.rst
Comment (pllim) [inline docs/development/workflow/maintainer_workflow.rst:182]:
```suggestion
the main branch. The logs are uploaded as a action artifact and can be viewed
```

File: docs/development/testguide.rst
Comment (pllim) [inline docs/development/testguide.rst:1080]:
```suggestion
Astropy also has the option to run a :ref:`comparative benchmark <benchmarks>` using Github Actions
```

**Review by pllim — COMMENTED** [review]
Just minor suggestions. I think this will be good to merge when Eero is satisfied. Thanks!

File: docs/development/workflow/maintainer_workflow.rst
Comment (MridulS) [reply docs/development/workflow/maintainer_workflow.rst:189]: No idea. This is taking me to the dark linguistics corners of the internet. It seems like either is fine. But flaky is more used now.

File: docs/development/testguide.rst
Comment (neutrinoceros) [inline docs/development/testguide.rst:1084]:
```suggestion
It is important to note that these benchmarks can be flaky as they run on
```

**Review by eerovaher — COMMENTED** [review]
I have a few remarks, mostly to improve consistency.

**Review by neutrinoceros — APPROVED** [review]

**Review by eerovaher — APPROVED** [review]

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**
